flow360/component/simulation/unit_system.py

```diff
@@ -217,12 +217,17 @@ def _unit_inference_validator(value, dim_name, is_array=False):
     -------
     unyt_quantity or value
     """
+
     if unit_system_manager.current:
         unit = unit_system_manager.current[dim_name]
         if is_array:
             if all(isinstance(item, Number) for item in value):
                 return value * unit
         if isinstance(value, Number):
+            if dim_name == "angle":
+                raise ValueError(
+                    "Angles must be provided in radians or degrees. Not as pure number."
```

Comment:
I think just the first sentence is good enough. Or "explicitly in radians or degrees"

Reply:
Removed. Andrzej introduced a new method.